example/glue/ObjectModelDelegate.hpp

```diff
@@ -201,6 +201,45 @@ class GC_ObjectModelDelegate
 		return false;
 	}
 
+	/**
+	 * Returns the field offset of the hottest field of the object referred to by the forwarded header.
+	 * Valid if scavenger dynamicBreadthFirstScanOrdering is enabled.
+	 *
+	 * @param forwardedHeader pointer to the MM_ForwardedHeader instance encapsulating the object
+	 * @return the offset of the hottest field of the given object referred to by the forwarded header, return U_8_MAX if a hot field does not exist
+	 */
+	MMINLINE U_8
+	getHotFieldOffset(MM_ForwardedHeader *forwardedHeader)
+	{
+		return U_8_MAX;
+	}
```

Comment:
I think in general OMR tries to use standard integer types instead of `U_8`.
```suggestion
	MMINLINE uint8_t
	getHotFieldOffset(MM_ForwardedHeader *forwardedHeader)
	{
		return UINT8_MAX;
	}
```

@amicic @jonoommen Should this glue API return a `uintptr_t` for field offsets?

Reply:
For field offsets I have used 'uint8' everywhere. I don't have a preference as both seem correct to me. Any thoughts @amicic?

Reply:
As far as the width is concerned we should be using the widest offset possible out of all downstream projects. OpenJ9 is the only one implementing this, and from what I can see the info comes from getInstanceFieldOffset() that returns uint32_t.

As far as using U_32 vs uint32_t (or any other width) we should be using the latter (I think it's the general trend in OMR code).

Reply:
Looking at this now after our offline discussion, shall I leave this as is? @youngar

Reply:
I'm fine with limiting the offset within uint8_t, since it's unknown if there would be any benefit with large field offsets.

As far as downstream projects are concerned, OpenJ9 while may have field offset larger than uint_8, for this particular purpose we made sure anything above that threshold will be ignored - so there is no correctness or safety issue.